language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java

```diff
@@ -187,7 +190,77 @@ protected boolean hasReadonlyKW(Node evalNodeAtCursor) {
                     .filter(varSymbol -> !existingFields.contains(varSymbol.getName().get()))
                     .collect(Collectors.toList());
             completionItems.addAll(this.getCompletionItemList(variables, context));
+            //Spread field can only be used with in a mapping constructor and the fields should be empty
+            if (existingFields.isEmpty() && evalNode.kind() == SyntaxKind.MAPPING_CONSTRUCTOR) {
+                completionItems.addAll(this.getSpreadFieldCompletionItems(evalNode, context));
+            }
+        }
+        return completionItems;
+    }
+
+    private List<LSCompletionItem> getSpreadFieldCompletionItems(Node evalNode, BallerinaCompletionContext context) {
+        ContextTypeResolver typeResolver = new ContextTypeResolver(context);
+        Optional<TypeSymbol> resolvedType = evalNode.apply(typeResolver);
+
+        if (resolvedType.isEmpty() || resolvedType.get().typeKind() != TypeDescKind.MAP) {
+            return Collections.emptyList();
         }
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition()).stream()
+                .filter(this.getVariableFilter().and(symbol -> {
+                    Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+                    if (typeDescriptor.isEmpty() || !typeDescriptor.get().subtypeOf(resolvedType.get())
+                            || (CommonUtil.getRawType(typeDescriptor.get()).typeKind() != TypeDescKind.MAP
+                            && CommonUtil.getRawType(typeDescriptor.get()).typeKind() != TypeDescKind.RECORD)) {
+                        return false;
+                    }
+                    return true;
+                })).collect(Collectors.toList());
+
+        return getSpreadFieldCompletionItemList(visibleSymbols, context);
+    }
+
+    private List<LSCompletionItem> getSpreadFieldCompletionItems(BallerinaCompletionContext context,
+                                                                 List<RecordFieldSymbol> validFields,
+                                                                 RecordTypeSymbol recordTypeSymbol) {
```

Comment:
Is this parameter `recordTypeSymbol` necessary?

Reply:
No it's not. Will remove it